beacon_node/network/src/sync/peer_sync_info.rs

```diff
@@ -27,82 +13,76 @@ pub enum PeerSyncType {
     Behind,
 }
 
-impl From<StatusMessage> for PeerSyncInfo {
-    fn from(status: StatusMessage) -> PeerSyncInfo {
-        PeerSyncInfo {
-            fork_digest: status.fork_digest,
-            finalized_root: status.finalized_root,
-            finalized_epoch: status.finalized_epoch,
-            head_root: status.head_root,
-            head_slot: status.head_slot,
-        }
-    }
-}
-
-impl Into<SyncInfo> for PeerSyncInfo {
-    fn into(self) -> SyncInfo {
-        SyncInfo {
-            status_head_slot: self.head_slot,
-            status_head_root: self.head_root,
-            status_finalized_epoch: self.finalized_epoch,
-            status_finalized_root: self.finalized_root,
+impl PeerSyncType {
+    pub fn as_sync_status(&self, info: &SyncInfo) -> PeerSyncStatus {
+        match self {
+            PeerSyncType::FullySynced => PeerSyncStatus::Synced { info: info.clone() },
+            PeerSyncType::Behind => PeerSyncStatus::Behind { info: info.clone() },
+            PeerSyncType::Advanced => PeerSyncStatus::Advanced { info: info.clone() },
         }
     }
 }
 
-impl PeerSyncInfo {
-    /// Derives the peer sync information from a beacon chain.
-    pub fn from_chain<T: BeaconChainTypes>(chain: &Arc<BeaconChain<T>>) -> Option<PeerSyncInfo> {
-        Some(Self::from(status_message(chain)?))
-    }
+pub fn remote_sync_type<T: BeaconChainTypes>(
+    local: &SyncInfo,
+    remote: &SyncInfo,
+    chain: &BeaconChain<T>,
+) -> PeerSyncType {
+    // auxiliary variables for clarity: Inclusive boundaries of the range in which we consider a peer's
+    // head "near" ours.
+    let near_range_start = local.head_slot - SLOT_IMPORT_TOLERANCE as u64;
```

Comment:
Around genesis are we going to underflow here?

Reply:
here we are using the `Sub` impl of `Slot` so it's saturating